### ISSUE TITLE
Spark unable to complete successfully against production loads.

### ISSUE THREAD

**Issue by naveedyahyazadeh** [jaegertracing/spark-dependencies#102]
I've been trying (and struggling) to get Spark working on a production deployment of Jaeger with an ES backend. I have tried reducing the look back window, ES_TIME_RANGE as low as 3 hours and I have assigned more than Spark's recommended CPU, memory, and JVM to the job.

In production, I expect about 25GB of data to be contained in a 3 hour window. When the spark job runs, ES's average cpu spikes to about 90% on some pods. Garbage Collection and heap utilization also spike during job execution.

The job itself ran for 6 hours and failed. The pod was deleted so I wasn't able to see logs. Spark's memory and CPU both idled at <1% for a majority of the test, then spiked to about 20% for the last hour of the test.

Is there any further configuration I can try to experiment with to help this job succeed?

**Comment by mehta-ankit** [comment]
FYI @pavolloffay

**Comment by mehta-ankit** [comment]
It seems like the issue was heap.
Once we assigned enough memory and increased heap the job was able to run.

we can share more details once we finish our tests and maybe something can be added to the README as suggestion.

**Comment by mehta-ankit** [comment]
`16 GB memory`, with `12 GB Heap` on the spark pods was what worked for us.
We had set the job to run every hour and look-back period (`ES_TIME_RANGE`) was set to 30m.
If we use more than 30m as look back time range, it puts a lot of load on our Elasticsearch cluster.

We had to find that sweet spot between our read request to ES cluster and resource needs of Spark to process that data.

We also enabled G1GC as the garbage collection algorithm and added some debug flags as JAVA_OPTS to find this issue with heap.
`-verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintGCApplicationStoppedTime`
